File: Problems/python/boj_에라토스테네스의체.py
#https://www.acmicpc.net/problem/2960
#Implementation

n, k = map(int, input().split())

def era(k):
    era = [True for _ in range(n+1)]
    for i in range(2, n+1):
        for j in range(i, n+1, i):
            if era[j]:
                era[j] = False
                k-=1
                if k==0:
                    print(j)
                    return
era(k)

File: Problems/cpp/boj_에라토스테네스의체.cpp
//https://www.acmicpc.net/problem/2960
//Implementation

#include <iostream>
#include <vector>

using namespace std;

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    int n, k;
    cin >> n >> k;
    vector<bool> era(n+1, false);

    for (int i=2; i<n+1; i++){
        for (int j=i; j<n+1; j=j+i){
            if (!era[j]) {
                era[j] = true;
                k--;

                if (k==0){
                    cout << j; 
                    break; 
                } 
            }
        }
    }
}